src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
                         {
-                            success = chain.Build(leafCert);
-                            AddChannelBindingToken(leafCert, arg);
+                            var serverCallback = easy._handler._serverCertificateValidationCallback;
+                            if (serverCallback == null)
+                            {
+                                SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(
+                                    chain, leafCert, checkCertName: false, hostName: null);
+                                success = errors == SslPolicyErrors.None;
```

Comment:
> Let libcurl do the validation when there's no callback, we do it when there is one.... If curl doing the validation always means a hostname match is blocked, we need to have them not do it when a user callback is presented, since the user could be forgiving the hostname violation

Yup, this is what's being done in the most recent update.  See the setting of VERIFYPEER and VERIFYHOST.